ros2bag/ros2bag/verb/convert.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import os
+
+from ros2bag.verb import VerbExtension
+
+
+class ConvertVerb(VerbExtension):
+    """ros2 bag convert."""
+
+    def add_arguments(self, parser, cli_name):  # noqa: D102
+        parser.add_argument(
+            'bag_file', help='bag file to convert')
+        parser.add_argument(
+            '-o', '--output',
+            help='destination of the bagfile to create, \
+            defaults to a timestamped folder in the current directory')
+        parser.add_argument(
+            '-s', '--storage', default='sqlite3',
```

Comment:
for a `convert` verb I would consider renaming this to `--input-storage` to highlight the conversion from `input` to `output`.

Reply:
I am not completely sure a default for `input` makes sense here. The storage identifier might be guessed from the `metadata.yaml` if present, but otherwise this has to be specified. 
It makes more sense for me to have it defaulted to `sqlite3` on the output side.

Reply:
In current master, the `play` and `info` verbs also set the input storage argument default to `sqlite3`, so I thought it best to keep the convention. I agree that it should be guessed, but the necessary api doesn't seem to be available in python, and IMO shouldn't be added in this PR.

I would rename it to `--input-storage`, remove the default and set required=True. I would also remove the short `-s` and not add `-i` to avoid confusion with `-o`, which is a filepath rather than storage specifier.